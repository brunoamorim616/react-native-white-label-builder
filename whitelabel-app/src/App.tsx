import React, {useState} from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-native-paper';
import Routes from './routes';
import Auth from './authentication';

export default function App() {
  const [auth, setAuth] = useState(false);

  return (
    <Provider>
      <NavigationContainer>{!auth ? <Auth /> : <Routes />}</NavigationContainer>
    </Provider>
  );
}
