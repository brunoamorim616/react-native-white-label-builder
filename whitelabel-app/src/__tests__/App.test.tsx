import 'react-native';
import React from 'react';
import {act} from 'react-test-renderer';
import {cleanup, render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';
import {App} from '../App';

// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

afterEach(cleanup);
describe('<Root/>', () => {
  test('Renders correctly', async () => {
    const tree = render(
      <NavigationContainer>
        <App />
      </NavigationContainer>,
    );
    await act(() => expect(tree).toMatchSnapshot());
  });
});
