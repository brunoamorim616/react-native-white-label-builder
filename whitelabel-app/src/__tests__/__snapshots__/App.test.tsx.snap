// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root/> Renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "width": 712.5,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View>
                <View
                  style={
                    [
                      {
                        "paddingBottom": 0,
                      },
                      {
                        "minHeight": 56,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                        ],
                        "zIndex": 3,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "backgroundColor": "rgba(255, 251, 254, 1)",
                          "bottom": 4,
                          "left": 10,
                          "maxHeight": 2,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                          "width": 12,
                        }
                      }
                    />
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "backgroundColor": "rgba(255, 251, 254, 1)",
                          "color": "transparent",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 8,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 0,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 33,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                            {
                              "scaleY": 1,
                            },
                          ],
                          "width": -16,
                          "writingDirection": "ltr",
                        }
                      }
                    >
                      Login
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        {
                          "color": "rgba(103, 80, 164, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-active"
                    >
                      Login
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-inactive"
                    >
                      Login
                    </Text>
                  </View>
                  <TextInput
                    accessibilityLabel="Text input field"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    selectionColor="rgba(103, 80, 164, 1)"
                    style={
                      [
                        {
                          "flexGrow": 1,
                          "margin": 0,
                          "zIndex": 1,
                        },
                        {
                          "height": 48,
                        },
                        {
                          "paddingBottom": 0,
                          "paddingTop": 0,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "paddingHorizontal": 16,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                      ]
                    }
                    testID="text-input-outlined"
                    underlineColorAndroid="transparent"
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "width": 712.5,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View>
                <View
                  style={
                    [
                      {
                        "paddingBottom": 0,
                      },
                      {
                        "minHeight": 56,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                        ],
                        "zIndex": 3,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "backgroundColor": "rgba(255, 251, 254, 1)",
                          "bottom": 4,
                          "left": 10,
                          "maxHeight": 2,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                          "width": 12,
                        }
                      }
                    />
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "backgroundColor": "rgba(255, 251, 254, 1)",
                          "color": "transparent",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 8,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 0,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 33,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                            {
                              "scaleY": 1,
                            },
                          ],
                          "width": -16,
                          "writingDirection": "ltr",
                        }
                      }
                    >
                      Password
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        {
                          "color": "rgba(103, 80, 164, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-active"
                    >
                      Password
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-inactive"
                    >
                      Password
                    </Text>
                  </View>
                  <TextInput
                    accessibilityLabel="Text input field"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    selectionColor="rgba(103, 80, 164, 1)"
                    style={
                      [
                        {
                          "flexGrow": 1,
                          "margin": 0,
                          "zIndex": 1,
                        },
                        {
                          "height": 48,
                        },
                        {
                          "paddingBottom": 0,
                          "paddingTop": 0,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "paddingHorizontal": 16,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                      ]
                    }
                    testID="text-input-outlined"
                    underlineColorAndroid="transparent"
                  />
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(103, 80, 164, 1)",
                      "borderColor": "transparent",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "minWidth": 64,
                      "width": 675,
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        {
                          "borderRadius": 20,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              {
                                "marginHorizontal": 24,
                                "marginVertical": 10,
                              },
                              undefined,
                              false,
                              {
                                "color": "rgba(255, 255, 255, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        Login
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
